### ISSUE TITLE
yolo tflite export fails for folders containing whitespace

### ISSUE THREAD

**Issue by Gornoka** [ultralytics/ultralytics#3850]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

### Bug

when exporting yolo to tflite on windows for a filepath that contains spaces the export fails. 

### Environment

Yolo: ultralytics 8.0.138
OS: Windows 

### Minimal Reproducible Example

````python
"""
minimal reproduction of yolo export failure
"""
import pathlib
from ultralytics import YOLO


def main():
    # for this to work you need to dump a yolo training result in the evil temp folder
    evil_path = pathlib.Path("D:/temp evil")
    best_models = list(evil_path.rglob("best.pt"))
    print(f"found models to convert: {best_models}")
    for model in best_models:
        print(f"converting model {str(model)}")
        model = YOLO(str(model))
        model.export(format="tflite")


if __name__ == "__main__":
    main()

````

### Additional

This fix would not be important normally, but unfortunately, MS decided in its infinite wisdom that Onedrives synced folders for Windows must have spaces in them, if you have both a company/edu and a personal Onedrive folder on your system.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@Gornoka thank you for bringing up this issue and your willingness to contribute a fix!

The problem arises, as you correctly identified, due to the white spaces in the file path. When the system interacts with such paths, it often misinterprets them, causing such issues.

Your workaround for this problem seems reasonable. However, I recommend taking a general approach when dealing with file and folder paths. Instead of manually managing file paths, you could leverage Python's built-in libraries like `os.path` or `pathlib` which are designed to handle these types of problems elegantly.

As for the issue with Microsoft Onedrive, unfortunately, this is something we can't change, but I take your point that it would make sense for our code to be robust against all sorts of pathnames.

Please feel free to submit a PR with your proposed changes. Be sure to include a detailed description of the changes and their impact, as well as tests demonstrating the correct behavior with your changes. I'm looking forward to reviewing your contribution to the YOLO community!

**Comment by Gornoka** [comment]
yeah i would love to get away from Onedrive. The path manipulation was already mostly based on Pathlib before, unfortunately the interactions of pathlib and subprocess.run&Popen is not that great.

**Comment by glenn-jocher** [comment]
@Gornoka hi there,

Thank you for your response. I understand the challenges posed by both the Onedrive path structure and the interaction between pathlib and the subprocess methods. You're absolutely right about these complications.

While pathlib provides an excellent utility for file and path manipulations, in some specific scenarios with specific functions, it may not function as expected. As you've identified, subprocess is one area where gaps can appear. These functions were not originally designed with pathlib Path objects in mind, resulting in potential issues with path handling.

In terms of working with subprocess specifically, one possible workaround could be converting pathlib Path objects to strings before passing them to the subprocess methods using the str() function.

This is certainly not an ideal situation in terms of code cleanliness and readability, but it might provide a temporary solution while we consider ways to improve path handling more broadly within our YOLOv8 codebase.

Please proceed with creating a Pull Request with the changes you see fit. We always appreciate community contributions to help make YOLOv8 the best it can be!

Best regards,
Glenn Jocher

**Comment by Gornoka** [comment]
> In terms of working with subprocess specifically, one possible workaround could be converting pathlib Path objects to strings before passing them to the subprocess methods using the str() function

The fixes from yesterday included that, I have decided against the cmd option with lists of parameters as I did not get it to work stable on Linux and Windows with the same code, and I did not want to add os dependent code if possible.

Kind regards
Lukas

**Comment by glenn-jocher** [comment]
@Gornoka hi Lukas,

Thanks for sharing your approach to address this issue. Your decision to avoid operating system-specific code solutions contributes to a more robust and maintainable codebase. Converting pathlib Path objects to strings before passing them to the subprocess methods seems like a practical solution given the constraints you're facing. We truly value your commitment to helping improve YOLOv8.

Best,
The YOLOv8 team

**Comment by glenn-jocher** [comment]
Great news 😃! Your original issue may now be resolved ✅ in PR #3879. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
